### ISSUE TITLE
Issue Postgres at height 239769

### ISSUE THREAD

**Issue by mvanhalen** [deso-protocol/core#553]
Postgres has an exception at block height 239769. We are out of sync.

`E0609 14:32:35.848315       1 blockchain.go:1384] MarkBlockInvalid: Block height: 239769, Block hash: 0000000000005bfa69662fe8276592e57289f47f65b2a7ec6436d4f6df622929, Error: ConnectBlock: error connecting txn #156: ConnectTransaction: : _connectAuthorizeDerivedKey: : _connectBasicTransferWithExtraSpend Problem spending balance: _spendBalance: amountNanos (736) exceeds spendable balance (0) at tipHeight (239768): RuleErrorInsufficientBalance
E0609 14:32:35.848338       1 blockchain.go:1385] MarkBlockInvalid: Printing stack trace so error is easy to find: 
E0609 14:32:35.848384       1 blockchain.go:1386] goroutine 1493 [running]:
runtime/debug.Stack()
    /usr/lib/go/src/runtime/debug/stack.go:24 +0x65
github.com/deso-protocol/core/lib.(*Blockchain).MarkBlockInvalid(0x19bc200?, 0xc0069b2f00, {0xc01e058ea0, 0x10e})
    /deso/src/core/lib/blockchain.go:1386 +0x12c
github.com/deso-protocol/core/lib.(*Blockchain).ProcessBlock(0xc20ed06000, 0xc006b616e0, 0xd0?)
    /deso/src/core/lib/blockchain.go:2394 +0x3425
github.com/deso-protocol/core/lib.(*Server)._handleBlock(0xc000a7a000, 0xc006177440, 0xc006b616e0)
    /deso/src/core/lib/server.go:1823 +0x785
github.com/deso-protocol/core/lib.(*Server)._handlePeerMessages(0xf4241?, 0xc0097f6d80?)
    /deso/src/core/lib/server.go:2142 +0x8d
github.com/deso-protocol/core/lib.(*Server).messageHandler(0xc000a7a000)
    /deso/src/core/lib/server.go:2187 +0x1bf
created by github.com/deso-protocol/core/lib.(*Server).Start
    /deso/src/core/lib/server.go:2375 +0xaf`

**Comment by mvanhalen** [comment]
@lazynina @AeonSw4n FYI seems heard others had a similar issue around this height as well.

**Comment by lazynina** [comment]
@mvanhalen - this looks like an issue w/ how the genesis block is stored in postgres. Will look into it.

**Comment by lazynina** [comment]
@mvanhalen - just published a new release that addresses the bug in how postgres processes the genesis block. Unfortunately, this requires a resync

**Comment by mvanhalen** [comment]
Thanks @lazynina  We are syncing with this update. We should hit this height later today / tomorrow

**Comment by StarGeezerPhil** [comment]
Thanks @mvanhalen for reporting the issue and @lazynina with a fix.

Any indication if this has resolved the issue? I'm worried about how long the last resync took.

**Comment by StarGeezerPhil** [comment]
@lazynina having reached block height 133859 my backend is now exiting, with nothing in the log:

```
I0621 12:14:00.884387       1 server.go:1749] Server._handleBlock: Received block ( 133857 / 243077 ) from Peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]
I0621 12:14:05.555821       1 server.go:1749] Server._handleBlock: Received block ( 133858 / 243077 ) from Peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]
I0621 12:14:06.292145       1 server.go:1749] Server._handleBlock: Received block ( 133859 / 243077 ) from Peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]
I0621 12:14:08.739741       1 base.go:301] Refreshing USD to ETH exchange rate
I0621 12:14:09.250422       1 base.go:290] Refreshing USD to BTC exchange rate
I0621 12:14:09.250446       1 base.go:308] New USD to ETH exchange rate: 1831.100000

I0621 12:14:13.123372       1 config.go:151] Logging to directory /db/v-00000
I0621 12:14:13.124188       1 config.go:152] Running node in MAINNET mode
Migrations already up to date
I0621 12:14:13.124229       1 config.go:153] Data Directory: /db/v-00000
```

On restart it commences as normal, fetches headers but then exits again before receiving all the bundles - each time exiting after fetching the header package for between ~161858 - 180000, e.g.

```
I0621 12:14:13.123372       1 config.go:151] Logging to directory /db/v-00000
I0621 12:14:13.124188       1 config.go:152] Running node in MAINNET mode
Migrations already up to date
I0621 12:14:13.124229       1 config.go:153] Data Directory: /db/v-00000
I0621 12:14:13.124236       1 config.go:160] Postgres URI: postgres://XXXYYYZZZ:XXXYYYZZZ@db:XYZ
I0621 12:14:13.124239       1 config.go:168] ForceChecksum: ON
I0621 12:14:13.124242       1 config.go:176] SnapshotBlockHeightPeriod: 1000
I0621 12:14:13.124246       1 config.go:180] ArchivalMode: ON
I0621 12:14:13.124249       1 config.go:183] SyncType: blocksync
I0621 12:14:13.124260       1 config.go:190] Connect IPs: [deso-seed-4.io]
I0621 12:14:13.124282       1 config.go:194] Add IPs: [deso-seed-4.io]
I0621 12:14:13.124288       1 config.go:202] READ ONLY MODE
I0621 12:14:13.124292       1 config.go:213] Max Inbound Peers: 125
I0621 12:14:13.124296       1 config.go:214] Protocol listening on port 17000
I0621 12:14:13.124302       1 config.go:220] Rate Limit Feerate: 0
I0621 12:14:13.124309       1 config.go:221] Min Feerate: 1000
badger 2023/06/21 12:14:26 INFO: All 1038 tables opened in 563ms
badger 2023/06/21 12:14:26 INFO: Discard stats nextEmptySlot: 0
badger 2023/06/21 12:14:26 INFO: Set nextTxnTs to 401580
I0621 12:14:29.794649       1 mempool.go:2420] Calling StartReadOnlyUtxoViewRegenerator...
I0621 12:14:29.794727       1 server.go:2373] Server.Start: Starting Server
I0621 12:14:29.853915       1 node.go:40] GlobalState BadgerDB Dir: /db/v-00000/badgerdb/global_state
I0621 12:14:29.854024       1 node.go:41] GlobalState BadgerDB ValueDir: /db/v-00000/badgerdb/global_state/badgerdb
I0621 12:14:30.151117       1 connection_manager.go:833] Full node socket initialized
badger 2023/06/21 12:14:30 INFO: All 1 tables opened in 35ms
badger 2023/06/21 12:14:30 INFO: Discard stats nextEmptySlot: 0
badger 2023/06/21 12:14:30 INFO: Set nextTxnTs to 1
badger 2023/06/21 12:14:30 INFO: Deleting empty file: /db/v-00000/badgerdb/global_state/badgerdb/000017.vlog
I0621 12:14:30.193666       1 base.go:246] Refreshing exchange rate...
I0621 12:14:32.402092       1 peer.go:1186] Peer.Start: Starting peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]
I0621 12:14:32.403560       1 peer.go:551] StartDeSoMessageProcessor: Starting for peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]
I0621 12:14:32.906816       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 133858 / 243115 ) total headers
W0621 12:14:32.907756       1 server.go:793] Server._handleHeaderBundle: Duplicate header 000000000000163d2a973a99962a87965effc353bb0c137c96b5021f6f002aca received from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ] in state SYNCING_HEADERS. Local header tip height 133858 hash 00000000000011c01ecba3873ce4d0934804d158e0e8deb90c8cdab387af88e9 with duplicate 000000000000163d2a973a99962a87965effc353bb0c137c96b5021f6f002aca
I0621 12:14:33.124347       1 base.go:185] Blockchain exchange rate: 1199 [1199 1199 1199 1199 1199 1199 1199 1199 1199 1199]
I0621 12:14:33.124422       1 base.go:250] Blockchain.com price (USD cents): 1199
I0621 12:14:33.402453       1 base.go:257] Coinbase price (USD Cents): 877.9999999999999
I0621 12:14:33.402495       1 base.go:286] Final exchange rate: 1199
I0621 12:14:33.402979       1 base.go:290] Refreshing USD to BTC exchange rate
I0621 12:14:33.707758       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 135858 / 243115 ) total headers
I0621 12:14:34.112367       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 137858 / 243115 ) total headers
I0621 12:14:34.383958       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 139858 / 243115 ) total headers
I0621 12:14:34.716544       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 141858 / 243115 ) total headers
I0621 12:14:35.065735       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 143858 / 243115 ) total headers
I0621 12:14:36.742652       1 base.go:297] New USD to BTC exchange rate: 28957.460000
I0621 12:14:36.742692       1 base.go:301] Refreshing USD to ETH exchange rate
I0621 12:14:40.463185       1 base.go:308] New USD to ETH exchange rate: 1831.080000
I0621 12:14:40.469632       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.470639       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.472689       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.472751       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.472778       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.472816       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.472831       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.472838       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.472849       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.474688       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.474748       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475276       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475326       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475336       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475347       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475356       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475364       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475377       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475385       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475427       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475439       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475523       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475534       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475549       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475557       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475563       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475570       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475579       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475896       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475940       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.475957       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.477911       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
I0621 12:14:40.478068       1 admin_fees.go:258] Error decoding desoOutputBytes to slice of DeSoOutputs: EOF - default to no additional fees
E0621 12:14:40.502018       1 verify.go:1266] SetJumioUSDCents: invalid bytes read: 0
E0621 12:14:40.502422       1 verify.go:1284] SetJumioKickbackUSDCents: invalid bytes read: 0
I0621 12:14:40.503218       1 server.go:2484] APIServer.Start: Starting APIServer
I0621 12:14:40.548433       1 server.go:2478] Listening to NON-SSL JSON API connections on port :17001
I0621 12:14:40.998782       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 145858 / 243115 ) total headers
I0621 12:14:41.280703       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 147858 / 243115 ) total headers
I0621 12:14:48.449306       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 149858 / 243115 ) total headers
I0621 12:14:49.045659       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 151858 / 243115 ) total headers
I0621 12:14:49.277954       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 153858 / 243115 ) total headers
I0621 12:14:49.746971       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 155858 / 243115 ) total headers
I0621 12:14:50.457668       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 157858 / 243115 ) total headers
I0621 12:14:50.480459       1 base.go:301] Refreshing USD to ETH exchange rate
I0621 12:14:50.480510       1 base.go:290] Refreshing USD to BTC exchange rate
I0621 12:14:50.486818       1 base.go:246] Refreshing exchange rate...
I0621 12:14:51.040160       1 base.go:308] New USD to ETH exchange rate: 1831.080000
I0621 12:14:51.208469       1 base.go:297] New USD to BTC exchange rate: 28964.900000
I0621 12:14:51.212136       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 159858 / 243115 ) total headers
I0621 12:14:52.170063       1 base.go:185] Blockchain exchange rate: 1199 [1199 1199 1199 1199 1199 1199 1199 1199 1199 1199]
I0621 12:14:52.170131       1 base.go:250] Blockchain.com price (USD cents): 1199
I0621 12:14:52.176741       1 server.go:769] Received header bundle with 2000 headers in state SYNCING_HEADERS from peer [ Remote Address: 35.232.92.5:17000 PeerID=1 ]. Downloaded ( 161858 / 243115 ) total headers
Migrations already up to date
I0621 12:14:52.336361       1 base.go:257] Coinbase price (USD Cents): 878.9999999999999
I0621 12:14:52.336779       1 base.go:286] Final exchange rate: 1199
I0621 12:14:57.665631       1 config.go:151] Logging to directory /db/v-00000
I0621 12:14:57.665947       1 config.go:152] Running node in MAINNET mode
```

I notice this line:
**35.232.92.5:17000 PeerID=1 ] in state SYNCING_HEADERS. Local header tip height 133858 hash 00000000000011c01ecba3873ce4d0934804d158e0e8deb90c8cdab387af88e9 with duplicate 000000000000163d2a973a99962a87965effc353bb0c137c96b5021f6f002aca**

**Comment by lazynina** [comment]
@StarGeezerPhil - I haven't been able to reproduce your issue unfortunately. Given that the node crashed without any error logs, I'm guessing there was some CPU or memory issue you hit on your machine. Will update you if I find any solutions, but this doesn't appear related to the issue causing postgres to mark blocks invalid at 239769

**Comment by StarGeezerPhil** [comment]
Thanks for looking @lazynina, I'll give the backend some more resources and see how it gets on. Only flag it here as the issue only started after this latest fix - it was fine before.

**Comment by StarGeezerPhil** [comment]
Just an update, in case it's helpful @lazynina @diamondhands0 

...I don't think we've been tracking this anywhere formally, but looks like a backend container now requires at least 16GB of RAM reserved - I've just retested the container at 8, 12 and 16 and it continued to fail on 8, on 12 it continued to sync but did exit after a few hours. 16 GB seems ok, but I've upped this to 20 GB to ensure some wiggle room. I suspect after sync it'll probably settle a little.
